src/Tools/dotnet-monitor/ParameterCapturing/EventParameterCapturingPipelineCache.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.WebApi.ParameterCapturing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.ParameterCapturing
+{
+    internal sealed class EventParameterCapturingPipelineCache
```

Comment:
See my comment in the `EventParameterCapturingPipeline`, this should become more of a builder-style class.

Reply:
I replaced the cache with `CapturedParametersBuilder`